src/continuous_agg.c

```diff
@@ -233,6 +264,68 @@ ts_materialization_invalidation_log_delete(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+static void
+continuous_agg_fill_bucket_function(int32 mat_hypertable_id, ContinuousAggsBucketFunction *bf)
+{
+	ScanIterator iterator;
+	int count = 0;
+
+	iterator = ts_scan_iterator_create(CONTINUOUS_AGGS_BUCKET_FUNCTION,
+									   AccessShareLock,
+									   CurrentMemoryContext);
+	init_scan_cagg_bucket_function_by_mat_hypertable_id(&iterator, mat_hypertable_id);
+	ts_scanner_foreach(&iterator)
+	{
+		Datum values[Natts_continuous_aggs_bucket_function];
+		bool isnull[Natts_continuous_aggs_bucket_function];
+
+		bool should_free;
+		HeapTuple tuple = ts_scan_iterator_fetch_heap_tuple(&iterator, false, &should_free);
+
+		/*
+		 * Our usual GETSTRUCT() approach doesn't work when TEXT fields are involved,
+		 * thus a more robust approach with heap_deform_tuple() is used here.
+		 */
+		heap_deform_tuple(tuple, ts_scan_iterator_tupledesc(&iterator), values, isnull);
+
+		Assert(!isnull[Anum_continuous_aggs_bucket_function_experimental - 1]);
```

Comment:
```suggestion
		Assert(!isnull[AttrNumberGetAttrOffset(Anum_continuous_aggs_bucket_function_experimental)]);
```

Reply:
@fabriziomello Thanks for the suggestion, but I'm afraid I can't accept it. Firstly, this would be inconsistent with the rest of the code (see the lines above). Secondly, this would be inconsistent with the PostgreSQL code style which we try to follow.

Reply:
@afiskon I agree It's an old style (dated from 1996) Postgres stuff used in few places:

```sh
main on  main [$?] 
➜ git grep AttrNumberGetAttrOffset
src/backend/catalog/index.c:                                 AttrNumberGetAttrOffset(atnum));
src/backend/replication/logical/relation.c:         attnum = AttrNumberGetAttrOffset(attnum);
src/include/access/attnum.h: * AttrNumberGetAttrOffset
src/include/access/attnum.h:#define AttrNumberGetAttrOffset(attNum) \
```

But I just suggested because it is used in a lot of places in our codebase:

```sh
timescaledb on  master [$!] 
➜ git grep AttrNumberGetAttrOffset | wc -l
414
```